tools/lsp/preview/native.rs

```diff
@@ -33,32 +33,55 @@ static GUI_EVENT_LOOP_STATE_REQUEST: LazyLock<Mutex<RequestedGuiEventLoopState>>
 
 thread_local! {static CLI_ARGS: std::cell::OnceCell<crate::Cli> = Default::default();}
 
-pub fn run_in_ui_thread<F: Future<Output = ()> + 'static>(
-    create_future: impl Send + FnOnce() -> F + 'static,
-) -> Result<(), String> {
-    // Wake up the main thread to start the event loop, if possible
-    {
-        let mut state_request = GUI_EVENT_LOOP_STATE_REQUEST.lock().unwrap();
-        if *state_request == RequestedGuiEventLoopState::Uninitialized {
-            *state_request = RequestedGuiEventLoopState::StartLoop;
-            GUI_EVENT_LOOP_NOTIFIER.notify_one();
-        }
-        // We don't want to call post_event before the loop is properly initialized
-        while *state_request == RequestedGuiEventLoopState::StartLoop {
-            state_request = GUI_EVENT_LOOP_NOTIFIER.wait(state_request).unwrap();
-        }
+fn fully_start_event_loop() -> Result<(), String> {
```

Comment:
maybe call it: `fully_start_ui_event_loop` to make clear this is about the slint event loop.  Or maybe `ensure_ui_event_loop`

Reply:
Changed and moved into the lsp_to_preview_message function;-)